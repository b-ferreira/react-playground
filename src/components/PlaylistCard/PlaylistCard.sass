.PlaylistCard
  width: $card-width
  height: $card-height
  cursor: pointer
  position: relative
  overflow: hidden
  margin: 5px

  @include tablet
    width: $card-width-tablet
    height: $card-height-tablet

  @include desktop
    margin: 14px
    width: $card-width-desktop
    height: $card-height-desktop

  &.not-playing
    .PlaylistCard__play
      transition: all .2s ease-in-out

    &:hover
      .PlaylistCard__image
        filter: blur(5px)

      .PlaylistCard__play
        opacity: 1

  &.playing
    .PlaylistCard__image
      filter: blur(5px)

    .PlaylistCard__play
      opacity: 1


.PlaylistCard__image
    max-width: $card-width
    max-height: $card-height

    @include tablet
      max-width: $card-width-tablet
      max-height: $card-height-tablet

    @include desktop
      max-width: $card-width-desktop
      max-height: $card-height-desktop

.PlaylistCard__play
    top: 0
    opacity: 0
    width: $card-width
    height: $card-height
    position: absolute
    background-color: rgba(0,0,0,0.5)

    @include tablet
      width: $card-width-tablet
      height: $card-height-tablet

    @include desktop
      width: $card-width-desktop
      height: $card-height-desktop

    span
      padding: .5rem 2rem .55rem


