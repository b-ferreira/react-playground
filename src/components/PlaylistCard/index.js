import React from 'react'
import PropTypes from 'prop-types'

const PlaylistCard = ({ playlist, onSelect, isPlaying = false }) => {
  const { images, name } = playlist

  return (
    <figure className={'PlaylistCard ' + (isPlaying ? 'playing' : 'not-playing animated')}
            onClick={() => onSelect(playlist)}>
      <img className="PlaylistCard__image" src={images[0].url} alt={name} />
      <div className="PlaylistCard__play flex flex-centered">
        <span className="br4 fc--default b bg--spotify">
          { isPlaying ? '| |' : 'Play' }
        </span>
      </div>
    </figure>
  )
}

PlaylistCard.displayName = 'PlaylistCard'
PlaylistCard.propTypes = {
  playlist: PropTypes.object.isRequired,
  onSelect: PropTypes.func.isRequired,
  isPlaying: PropTypes.bool
}

export default PlaylistCard
