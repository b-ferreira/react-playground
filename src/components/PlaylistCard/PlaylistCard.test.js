import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import PlaylistCard from './index'

const playlistMock = {
  name: 'Awesome Playlist',
  images: [
    {
      url: 'asd'
    }
  ]
}
const selectCallback = jest.fn()

it('renders correctly', () => {
  const tree = renderer
    .create(<PlaylistCard playlist={playlistMock} onSelect={selectCallback} />)
    .toJSON()
    expect(tree).toMatchSnapshot()
})

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <PlaylistCard playlist={playlistMock} onSelect={selectCallback} />,
    div
  )
  ReactDOM.unmountComponentAtNode(div)
})
