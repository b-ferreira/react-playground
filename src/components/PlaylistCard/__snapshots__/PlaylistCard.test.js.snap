// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<figure
  className="PlaylistCard not-playing animated"
  onClick={[Function]}
>
  <img
    alt="Awesome Playlist"
    className="PlaylistCard__image"
    src="asd"
  />
  <div
    className="PlaylistCard__play flex flex-centered"
  >
    <span
      className="br4 fc--default b bg--spotify"
    >
      Play
    </span>
  </div>
</figure>
`;
